NERC-CEH/plankton_ml
# Quick attempt at model interpretation

Considered leaving this on a branch as the results were so inconclusive, but there's some useful utils and improved test coverage in here as well that seem worth a merge - would appreciate eyes on it.

## What's in here

* Tiny bit of refactoring for ease of loading the `scivision` model either with or without its final layer (the classification one)
* Associated tests, increased overall coverage
* A notebook which leans heavily on the Captum tutorial walkthroughs, applying a range of methods to heatmap areas of plankton images which have influenced the predicted class

The visualisations don't look very meaningful to me, wrote a bit about probable reasons why, and what's worth doing next, in #6   

## To test

* run the python tests as per the README (minimum: `export PYTHONPATH=.; py.test`
* if you've run through the outstanding PR in https://github.com/NERC-CEH/plankton_ml/pull/5 then you'll have the chromadb index with the images and embeddings in it
* The notebook should be a matter of `Run all` - it needs the location of the s3 object store set in `.env` - see also the README
